model.py

```diff
@@ -476,316 +450,153 @@ def service_area_name(self):
+        ne_ref_pt_geom     = cast(northeast_ref_pt, Geometry)   # noqa: E221
+
+        # This MUST have two Geometry parameters. If it uses Geography the distance returned
+        # will be in meters, NOT radians.
+        local_dist_in_rads = func.ST_Distance(target_geometry, ne_ref_pt_geom)
+
+        # Step 2: Use the localized value to set up a geometry-based radius function
+        #         that's locally accurate (for rough values of 'accurate'). This will
+        #         let us limit our search to Places no further than max_radius km from
+        #         our target location.
+        within_max_radius = func.ST_DWithin(target_geometry, Place.geometry, local_dist_in_rads)
+
+        # Step 3: Set up a function to find the minimum distance between the
+        #         target (a point), and an edge of the Place objects we want
+        #         to search, which are typically polygons. We won't restrict
+        #         the query on this, but we do want to return it as a column,
+        #         and use it to order results.
+        meters_to_near_edge = func.min(func.ST_DistanceSphere(target_geometry, Place.geometry))
+
+        # Step 4: Establish the query object, and restrict it to the right feed.
+        #         We only want to search against the geometry of Place objects which
+        #         are used as the service area of one or more Libraries, to avoid
+        #         searching against all Places loaded in the database, so we join
+        #         Library -> ServiceArea -> Place.
+        query_obj = db_session.query(Library).join(Library.service_areas).join(ServiceArea.place)
+        query_obj = query_obj.filter(cls._feed_restriction(production))
+
+        # Step 5: Use our radius function to add a WHERE clause, so that we return
+        #         only Library objects within the max_radius in km.
+        query_obj = query_obj.filter(within_max_radius)
+
+        # Step 6: Add the distance from our target point to the near edge as a
+        #         column in the output, group our results by Library, and then
+        #         use the point-to-polygon-edge distance to order our result set.
+        query_obj = query_obj.add_columns(meters_to_near_edge)
+        query_obj = query_obj.group_by(Library.id)
+        query_obj = query_obj.order_by(meters_to_near_edge.asc())
+
+        return query_obj
 
-        # Decrease the score based on the sum of the sizes of the library's focus areas, in km^2.
-        # This currently  assumes that the library's focus areas don't overlap, which may not be true.
-        # If a focus area is "everywhere", the size is the area of Earth (510 million km^2).
-        focus_area_size = func.sum(
-            case(
-                [(focus_areas_subquery.c.type==Place.EVERYWHERE, literal_column(str(510000000000000)))],
-                else_=func.ST_Area(focus_areas_subquery.c.geometry)
-            )
-        ) / 1000000
```

Comment:
Since we are handling supralocal libraries separately I'm OK with removing this.